pkg/webhook/resources/upgrade/validator.go

```diff
@@ -100,7 +108,20 @@ func (v *upgradeValidator) Create(request *types.Request, newObj runtime.Object)
 		return werror.NewBadRequest("there are degraded volumes, please check all volumes are healthy")
 	}
 
-	return nil
+	cluster, err := v.clusters.Get(util.FleetLocalNamespaceName, util.LocalClusterName)
+	if err != nil {
+		return werror.NewInternalError(fmt.Sprintf("can't find %s/%s cluster, err: %+v", util.FleetLocalNamespaceName, util.LocalClusterName, err))
+	}
+
+	if cluster.Status.Phase != string(clusterv1alpha4.ClusterPhaseProvisioned) {
+		return werror.NewBadRequest(fmt.Sprintf("cluster %s/%s status is provisioning, please wait for it to be provisioned", util.FleetLocalNamespaceName, util.LocalClusterName))
```

Comment:
better to replace `provisioning` with `string(cluster.Status.Phase)` to reflect the actual phase

Reply:
Updated it. Thanks.